sdk/src/Services/DynamoDBv2/Custom/Conversion/SchemaV1.cs

```diff
@@ -416,48 +544,25 @@ protected override bool TryFrom(DynamoDBBool b, Type targetType, out bool result
             result = b.Value;
             return true;
         }
+#if NET8_0_OR_GREATER
+        protected override bool TryFrom(Primitive p, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.PublicConstructors)] Type targetType, out bool result)
+#else
         protected override bool TryFrom(Primitive p, Type targetType, out bool result)
+#endif
         {
             result = !p.StringValue.Equals("0", StringComparison.OrdinalIgnoreCase);
             return true;
         }
     }
 
-#if NET8_0_OR_GREATER
-    [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode(Amazon.DynamoDBv2.Custom.Internal.InternalConstants.RequiresUnreferencedCodeMessage)]
-#endif
     internal abstract class CollectionConverter : Converter
     {
-        private IEnumerable<Type> targetTypes;
-        private static IEnumerable<Type> GetTargetTypes(IEnumerable<Type> memberTypes)
```

Comment:
This method was doing lots of reflection to expand all of the "primitive" types into the varies array, list and set variations. This unbounded reflection is not possible in reflection. I replaced the logic by creating a new static list of types with all of the "primitive" permeations.